app/javascript/packs/suspendUserModal.js

```diff
@@ -0,0 +1,103 @@
+import { showWindowModal } from '@utilities/showModal';
+import { request } from '@utilities/http';
+
+const suspendUser = async ({ event, userId, username, suspensionReason }) => {
+  event.preventDefault();
+  closeSuspendUserModal();
+
+  try {
+    const response = await request(
+      `/admin/member_manager/users/${userId}/user_status`,
+      {
+        method: 'PATCH',
+        body: JSON.stringify({
+          id: userId,
+          user: {
+            note_for_current_role: suspensionReason,
+            user_status: 'Suspend',
+          },
+        }),
+        credentials: 'same-origin',
+      },
+    );
+
+    const outcome = await response.json();
+
+    /* eslint-disable no-restricted-globals */
+    if (outcome.success) {
+      top.addSnackbarItem({
+        message: `Success: "${username}" has been suspended.`,
+        addCloseButton: true,
+      });
+    } else {
+      top.addSnackbarItem({
+        message: `Error: something went wrong; ${username} NOT suspended.`,
+        addCloseButton: true,
+      });
+    }
+  } catch (error) {
+    top.addSnackbarItem({
+      message: `Error: ${error}`,
+      addCloseButton: true,
+    });
+  }
+};
+
+function closeSuspendUserModal() {
```

Comment:
I'd like to explain this function:

When I previously imported and called `closeWindowModal` to close the Suspend User modal, it did not fire (the modal did not close).

So, in the `closeWindowModal` function, I passed in a default value for `document`, so that in my `suspenduserModal` file, I could use `window.parent.document` for my `document`. Then the Suspend User modal closed successfully:
```javascript
export const closeWindowModal = async ({ document = window.document }) => {
  const currentModalContainer = document.getElementById(WINDOW_MODAL_ID);
  if (currentModalContainer) {
    const { render } = await getPreactImport();
    render(null, currentModalContainer);
  }
};
``` 

HOWEVER, Cypress tests broke in three places, claiming that they couldn't call `document` on `undefined`; basically, the tests could not find `window`. I tried different things to get it to work, but nothing did.

So I decided on the (albeit hacky) solution of locating and clicking the Suspend User close-button programatically.

I'm all ears for a better solution. 😓

Reply:
Ah, OK so the problem is if we change the `closeWindowModal` signature to:

`export const closeWindowModal = async ({ document = window.document })`

The curly brackets around `document = window.document` are trying to [destructure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) the properties of an object. Current uses of `closeWindowModal` are not passing any object, they're just calling `closeWindowModal()`. So we get an error thrown because what is happening is equivalent to:

```js
export const closeWindowModal = async (props) => {
  const document = props.document;
}
```

But `props` is `undefined`, and we get this error "Cannot read properties of undefined".

Given we are not likely to need multiple options passed to `closeWindowModal` I would suggest a small tweak to get this working:

```js
export const closeWindowModal = async (document = window.document) => {
  const currentModalContainer = document.getElementById(WINDOW_MODAL_ID);
  if (currentModalContainer) {
    const { render } = await getPreactImport();
    render(null, currentModalContainer);
  }
};
```

and then to call it from `suspendUserModal`:

```js
function closeSuspendUserModal() {
  closeWindowModal(window.parent.document);
}
```

The only difference is we don't expect/pass an object to `closeWindowModal`, meaning all the current uses of `closeWindowModal()` won't need to be changed 🙂

Reply:
@aitchiss Thank you for explaining this so clearly! I got it: I was passing an Object, so the function was looking for something to destruct. 😓